lib/logstash-docket/aliased_plugin.rb

```diff
@@ -0,0 +1,85 @@
+# encoding: utf-8
+
+require_relative 'plugin'
+
+module LogstashDocket
+
+  class AliasPlugin
+
+    SUPPORTED_TYPES = Set.new(%w(input output filter codec).map(&:freeze)).freeze
+
+    include Plugin
+
+    attr_reader :canonical_plugin
+
+    def initialize(canonical_plugin:, alias_name:)
+      fail(ArgumentError) unless canonical_plugin.kind_of?(ArtifactPlugin)
+
+      super(type: canonical_plugin.type, name: alias_name)
+
+      fail("#{canonical_plugin.desc} plugin type #{type} not supported as an integration plugin") unless SUPPORTED_TYPES.include?(type)
+
+      @canonical_plugin = canonical_plugin
+    end
+
+    ##
+    # @see Plugin#version
+    def version
+      @canonical_plugin.version
+    end
+
+    def documentation
+      content = @canonical_plugin.documentation
+
+      # TODO: control the hard codings through central management for the long term
+      # we had a discussion with Ry, we will manipulate these params through logstash repo AliasRegistry.yml
+      # this will be covered by https://github.com/elastic/docs-tools/issues/69
```

Comment:
@yaauie as we talked offline, created a new issue #69 to handle hard codings here as well as checksum check in `AliasRegistry.java` of the logstash repo.